csharp/src/Ice/InputStream.cs

```diff
@@ -9,14 +9,6 @@
 
 namespace Ice;
 
-/// <summary>
-/// Throws a UserException corresponding to the given Slice type Id, such as "::Module::MyException".
-/// If the implementation does not throw an exception, the Ice run time will fall back
-/// to using its default behavior for instantiating the user exception.
-/// </summary>
-/// <param name="id">A Slice type Id corresponding to a Slice user exception.</param>
-public delegate void UserExceptionFactory(string id);
```

Comment:
We never used this delegate anywhere. I believe it was provided for consistency with C++, where it's used by the scripting language extensions. It's likely we can use the regular Slice loader for exceptions in C++ as well.